weasyl/searchtag.py

```diff
@@ -167,7 +167,7 @@ def is_tag_restriction_pattern_valid(text):
     return False
 
 
-def associate(userid, tags, submitid=None, charid=None, journalid=None):
+def associate(userid, tags, submitid=None, charid=None, journalid=None, artistid=None, optoutid=None):
```

Comment:
I don't disagree but refactoring get_targetid and related is probably an issue all on its own.

point on name changes taken